forc-plugins/forc-fmt/src/main.rs

```diff
@@ -63,35 +76,178 @@ fn run() -> Result<()> {
+    let mut formatter = Formatter::from_dir(dir)?;
+    let mut members = vec![];
+
+    for member_path in workspace.member_paths()? {
+        members.push(member_path)
+    }
+
+    // Format files at the root - we do not want to start calling format_pkg_at_dir() here,
+    // since this would mean we format twice on each subdirectory.
+    if let Ok(read_dir) = fs::read_dir(dir) {
+        for entry in read_dir.filter_map(|res| res.ok()) {
+            let path = entry.path();
+            if path.is_file() && is_sway_file(&path) {
+                format_file(
+                    app,
+                    path,
+                    Some(workspace.dir().to_path_buf()),
+                    &mut formatter,
+                )?;
+            }
+        }
+    }
+
+    // Format subdirectories. We do not call format on members directly here, since
+    // in workspaces, it is perfectly valid to have subdirectories containing Sway files,
+    // yet not be a member of the workspace.
+    for sub_dir in get_sway_dirs(dir.to_path_buf()) {
+        if sub_dir.join(constants::MANIFEST_FILE_NAME).exists() {
+            // Here, we cannot simply call Formatter::from_dir() and rely on defaults
+            // if there is no swayfmt.toml in the sub directory because we still want
+            // to use the swayfmt.toml at the workspace root (if any).
+            // In order of priority: member > workspace > default.
+            formatter = Formatter::from_dir(&sub_dir)?;
+        }
+        format_pkg_at_dir(app, &sub_dir, &mut formatter)?;
+    }
+
+    let manifest_file = dir.join(constants::MANIFEST_FILE_NAME);
+
+    // Finally, format the root manifest using taplo formatter
+    if let Ok(edited) = format_manifest(app, manifest_file) {
+        contains_edits = edited;
+    }
+
+    if app.check {
+        if contains_edits {
+            // One or more files are not formatted, exit with error
+            bail!("Files contain formatting violations.");
+        } else {
+            // All files are formatted, exit cleanly
+            Ok(())
+        }
+    } else {
+        Ok(())
+    }
+}
+
+/// Format the given manifest at a path.
+/// Returns:
+/// - Ok(true) if executed successfully and formatted,
+/// - Ok(false) if executed successfully and not formatted,
+/// - Err if it fails to execute at all.
+fn format_manifest(app: &App, manifest_file: PathBuf) -> Result<bool> {
+    if let Ok(manifest_content) = fs::read_to_string(&manifest_file) {
+        let mut edited = false;
+        let taplo_alphabetize = taplo_fmt::Options {
+            reorder_keys: true,
+            ..Default::default()
+        };
+        let formatted_content = taplo_fmt::format(&manifest_content, taplo_alphabetize);
+        if !app.check {
+            write_file_formatted(&manifest_file, &formatted_content)?;
+        } else if formatted_content != manifest_content {
+            edited = true;
+            error!("\nManifest Forc.toml improperly formatted");
```

Comment:
Is this really an error? Seems like a warning/info to me

```suggestion
            error!("Improperly formatted manifest file: {}", manifest_file);
```

Reply:
This is if the formatter was run with `--check` i.e. `app.check == true`, which should error out if the manifest is not properly formatted.